src/main/java/io/temporal/samples/interceptor/workflow/MyWorkflowImpl.java

```diff
@@ -0,0 +1,60 @@
+/*
+ *  Copyright (c) 2020 Temporal Technologies, Inc. All Rights Reserved
+ *
+ *  Copyright 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not
+ *  use this file except in compliance with the License. A copy of the License is
+ *  located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed on
+ *  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package io.temporal.samples.interceptor.workflow;
+
+import io.temporal.workflow.Workflow;
+import java.time.Duration;
+
+public class MyWorkflowImpl implements MyWorkflow {
+
+  private String name;
+  private String title;
+
+  @Override
+  public String exec() {
+
+    // wait for a greeting
+    Workflow.await(() -> name != null && title != null);
+
+    // execute child workflow
+    MyChildWorkflow child = Workflow.newChildWorkflowStub(MyChildWorkflow.class);
+    String result = child.execChild(name, title);
+
+    Workflow.sleep(Duration.ofSeconds(2));
```

Comment:
Do we need this sleep? Is this sleep the reason why do we wait or 2500ms in InterceptorStarter? In that case, maybe we can just wait for the end of the workflow?

Reply:
Yeah I'll change to Workflow.await for exit signal it will be easier to understand.

Reply:
done